api/src/main/java/org/openmrs/module/fhirExtension/validators/DiagnosticReportBasedOnValidator.java

```diff
@@ -0,0 +1,46 @@
+package org.openmrs.module.fhirExtension.validators;
+
+import ca.uhn.fhir.rest.server.exceptions.UnprocessableEntityException;
+import org.hl7.fhir.r4.model.DiagnosticReport;
+import org.hl7.fhir.r4.model.OperationOutcome;
+import org.openmrs.Order;
+import org.openmrs.Patient;
+import org.openmrs.api.OrderService;
+import org.openmrs.module.fhir2.model.FhirDiagnosticReport;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class DiagnosticReportBasedOnValidator {
+	
+	static final String INVALID_ORDER_ERROR_MESSAGE = "Given lab order is not prescribed by the doctor";
+	
+	@Autowired
+	private OrderService orderService;
+	
+	public void validate(FhirDiagnosticReport fhirDiagnosticReport, DiagnosticReport diagnosticReport) {
```

Comment:
Validator should have no knowledge about DiagnosticReport since its only validating FhirDiagnosticReport